pkg/querier/queryrange/split_by_interval_test.go

```diff
@@ -440,6 +440,377 @@ func Test_splitByInterval_Do(t *testing.T) {
+								{Timestamp: time.Unix(0, 3*time.Hour.Nanoseconds()-1), Line: fmt.Sprintf("%d", 3*time.Hour.Nanoseconds()-1)},
+								{Timestamp: time.Unix(0, 3*time.Hour.Nanoseconds()), Line: fmt.Sprintf("%d", 3*time.Hour.Nanoseconds())},
+								{Timestamp: time.Unix(0, 4*time.Hour.Nanoseconds()-1), Line: fmt.Sprintf("%d", 4*time.Hour.Nanoseconds()-1)},
+							},
+						},
+					},
+				},
+			},
+		},
+
+		{
+			"forward limited",
+			&LokiRequest{
+				StartTs:   time.Unix(0, (5 * time.Minute).Nanoseconds()),
+				EndTs:     time.Unix(0, (4 * time.Hour).Nanoseconds()),
+				Query:     "",
+				Limit:     2,
+				Step:      1,
+				Direction: logproto.FORWARD,
+				Path:      "/api/prom/query_range",
+			},
+			&LokiResponse{
+				Status:    loghttp.QueryStatusSuccess,
+				Direction: logproto.FORWARD,
+				Limit:     2,
+				Version:   1,
+				Data: LokiData{
+					ResultType: loghttp.ResultTypeStream,
+					Result: []logproto.Stream{
+						{
+							Labels: `{foo="bar", level="debug"}`,
+							Entries: []logproto.Entry{
+								{Timestamp: time.Unix(0, time.Hour.Nanoseconds()-1), Line: fmt.Sprintf("%d", time.Hour.Nanoseconds()-1)},
+								{Timestamp: time.Unix(0, time.Hour.Nanoseconds()), Line: fmt.Sprintf("%d", time.Hour.Nanoseconds())},
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			"backward limited",
+			&LokiRequest{
+				StartTs:   time.Unix(0, (5 * time.Minute).Nanoseconds()),
+				EndTs:     time.Unix(0, (4 * time.Hour).Nanoseconds()),
+				Query:     "",
+				Limit:     2,
+				Step:      1,
+				Direction: logproto.BACKWARD,
+				Path:      "/api/prom/query_range",
+			},
+			&LokiResponse{
+				Status:    loghttp.QueryStatusSuccess,
+				Direction: logproto.BACKWARD,
+				Limit:     2,
+				Version:   1,
+				Data: LokiData{
+					ResultType: loghttp.ResultTypeStream,
+					Result: []logproto.Stream{
+						{
+							Labels: `{foo="bar", level="debug"}`,
+							Entries: []logproto.Entry{
+								{Timestamp: time.Unix(0, 3*time.Hour.Nanoseconds()), Line: fmt.Sprintf("%d", 3*time.Hour.Nanoseconds())},
+								{Timestamp: time.Unix(0, 4*time.Hour.Nanoseconds()-1), Line: fmt.Sprintf("%d", 4*time.Hour.Nanoseconds()-1)},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			res, err := split.Do(ctx, tt.req)
+			require.NoError(t, err)
+			require.Equal(t, tt.want, res)
+		})
+	}
+}
+
+func Test_alignedIntervals(t *testing.T) {
+	l := WithDefaultLimits(fakeLimits{}, queryrange.Config{SplitQueriesByInterval: time.Hour})
+	// next := queryrange.HandlerFunc(func(_ context.Context, r queryrange.Request) (queryrange.Response, error) {
+	// 	return &LokiResponse{}, nil
+	// })
+
+	// split := &splitByInterval{
+	// 	next:     next,
+	// 	limits:   l,
+	// 	merger:   LokiCodec,
+	// 	metrics:  nilMetrics,
+	// 	splitter: splitByTime,
+	// }
+
+	tests := []struct {
```

Comment:
Actually seems like this test is enough. it should cover your change. I guess we can remove the other unless I missed something.

Reply:
I thought there would be some value in having both, but I can see how the other test would be confusing.